### PR TITLE
refactor: switched onboarding/profile to using genericfield

### PR DESCRIPTION
**Pull request by mikewolfd**

### CODE REVIEW

File: web/src/components/GenericNumericField.tsx [hunk 1]
```diff
@@ -1,43 +1,12 @@
...
-      value={value.toString()}
-      additionalValidation={additionalValidation}
-      visualFilter={props.visualFilter ? props.visualFilter : regex}
-      fieldOptions={{
-        inputProps: { inputMode: "numeric" },
-      }}
-    />
-  );
+export const GenericNumericField = ({ minLength, maxLength, ...props }: Props): ReactElement => {
+  return <GenericTextField {...props} numericProps={{ minLength, maxLength }} />;
```
Comment (aloverso) [inline web/src/components/GenericNumericField.tsx:11]: what is the purpose of this component? It just seems like a passthrough. Why not just have us pass numeric props to GenericTextField when we need a numeric field?

File: web/src/components/onboarding/NumericField.tsx [hunk 2]
```diff
@@ -1,53 +1,19 @@
...
-
-    props.onValidation(props.fieldName, !valid);
-  };
-
-  const handleChange = (): void => props.onValidation(props.fieldName, false);
-
-  const regex = (value: string): string => value.replace(/[^0-9]/g, "");
+interface Props extends Omit<OnboardingProps, "numericProps">, NumericFieldProps {}
 
+export const NumericField = ({ minLength, maxLength, ...props }: Props): ReactElement => {
```
Comment (aloverso) [inline web/src/components/onboarding/NumericField.tsx:9]: if this is specific to onboarding we should probably call it OnboardingNumericField

Comment (aloverso) [reply web/src/components/onboarding/NumericField.tsx:9]: but also again, now that it doesn't have any of its own logic, why does this component need to exist at all?

File: web/src/components/tasks/business-formation/BusinessFormationNumericField.tsx [hunk 3]
```diff
@@ -1,41 +1,16 @@
...
-      <Content>{state.displayContent[props.fieldName].contentMd}</Content>
-      <GenericNumericField
-        value={state.formationFormData[props.fieldName]}
-        placeholder={state.displayContent[props.fieldName].placeholder}
-        error={state.errorMap[props.fieldName].invalid}
-        handleChange={handleChange}
-        onValidation={onValidation}
-        {...props}
-      />
+      <BusinessFormationTextField {...props} numericProps={{ minLength, maxLength }} />
```
Comment (aloverso) [inline web/src/components/tasks/business-formation/BusinessFormationNumericField.tsx:13]: doesn't seem like this component needs to exist

File: web/test/pages/profile.test.tsx [hunk 4]
```diff
@@ -184,7 +184,7 @@ describe("profile", () => {
     clickSave();
     await waitFor(() => {
       expect(
-        subject.queryByText(templateEval(OnboardingDefaults.errorTextMinimumNumericField, { length: "10" }))
+        subject.queryByText(templateEval(OnboardingDefaults.errorTextMinimumNumericField, { length: "9" }))
```
Comment (aloverso) [inline web/test/pages/profile.test.tsx:187]: why did this change?

Comment (mikewolfd) [reply web/test/pages/profile.test.tsx:187]: How it calculated the length in reference to the hyphen within the field changed, this is more accurate.

**Pull request closed without merging**